### PR TITLE
Put agent info batch does not require a space

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Addresses a tech debt item that was raised here -> https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs#L920

### TODO:
- [x] Depends on #3540
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs [hunk 1]
```diff
@@ -922,19 +922,16 @@ impl MetaNet {
                                 // the internal peer space was used instead we
                                 // could do this in a single event with the
                                 // whole list.
-                                for peer in peer_list {
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs:925]: delete the todo?

**Review by neonphog — APPROVED** [review]
:+1:

File: crates/holochain_p2p/src/spawn/actor.rs [hunk 2]
```diff
@@ -638,14 +638,29 @@ impl kitsune_p2p::event::KitsuneP2pEventHandler for HolochainP2pActor {
...
+                .fold(
+                    HashMap::<DnaHash, Vec<AgentInfoSigned>>::new(),
+                    |mut acc, (dna, agent)| {
+                        acc.entry(dna).or_default().push(agent);
+                        acc
+                    },
+                );
+
+            for (dna, agents) in put_requests {
+                evt_sender.put_agent_info_signed(dna, agents).await?;
```
Comment (jost-s) [inline crates/holochain_p2p/src/spawn/actor.rs:657]: Can putting be done in parallel if that makes sense?

**Review by jost-s — APPROVED** [review]

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs (hunk 1 above)
Comment (ThetaSinner) [reply crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs:925]: Oops, yes definitely

File: crates/holochain_p2p/src/spawn/actor.rs (hunk 2 above)
Comment (ThetaSinner) [reply crates/holochain_p2p/src/spawn/actor.rs]: It can, just requires a rather ugly block of code - done :)

Comment (jost-s) [reply crates/holochain_p2p/src/spawn/actor.rs]: Amazingly ugly indeed!

**Pull request merged**